rust/openvasd/src/controller/mod.rs

```diff
@@ -48,27 +59,112 @@ pub enum ClientIdentifier {
     /// subject of a known client certificate. Based on that we don't need more information.
     Known(ClientHash),
 }
-/// Is used to transfer the information if there is an identifier present within the connection
-pub trait ClientGossiper {
-    /// Gets the identifier
-    ///
-    /// Based on the concurrent nature, the actual information is boxed within a Arc and locked for
-    /// concurrent read write accesses.
-    fn client_identifier(&self) -> &std::sync::Arc<std::sync::RwLock<ClientIdentifier>>;
-}
 
 impl<T> Scanner for T where T: ScanStarter + ScanStopper + ScanDeleter + ScanResultFetcher {}
 
+fn retrieve_and_reset(id: Arc<RwLock<ClientIdentifier>>) -> ClientIdentifier {
+    // get client information
+    let mut ci = id.write().unwrap();
+    let cci = ci.clone();
+    // reset client information
+    *ci = ClientIdentifier::Unknown;
+    drop(ci);
+    cci
```

Comment:
why overwrite `ci` and drop it directly afterwards manually? Isn't it drop regardless after the end of the function?

Reply:
the handler call may be slow. to remove the lock I drop it immediately so that the next request doesn't have to wait until the handling of the previous request is done.

Reply:
ah nevermind, wrong function. yes, you're right I just drop out of habit.